core/sequencer/sequencer.go

```diff
@@ -97,52 +95,70 @@ func New(mapID int64,
 // Initialize inserts the object hash of an empty struct into the log if it is empty.
 // This keeps the log leaves in-sync with the map which starts off with an
 // empty log root at map revision 0.
-func (s *Sequencer) Initialize(ctx context.Context) error {
+func (s *Sequencer) Initialize(ctx context.Context, logID, mapID int64) error {
 	logRoot, err := s.tlog.GetLatestSignedLogRoot(ctx, &trillian.GetLatestSignedLogRootRequest{
-		LogId: s.logID,
+		LogId: logID,
 	})
 	if err != nil {
-		return fmt.Errorf("GetLatestSignedLogRoot(%v): %v", s.logID, err)
+		return fmt.Errorf("GetLatestSignedLogRoot(%v): %v", logID, err)
 	}
 	mapRoot, err := s.tmap.GetSignedMapRoot(ctx, &trillian.GetSignedMapRootRequest{
-		MapId: s.mapID,
+		MapId: mapID,
 	})
 	if err != nil {
-		return fmt.Errorf("GetSignedMapRoot(%v): %v", s.mapID, err)
+		return fmt.Errorf("GetSignedMapRoot(%v): %v", mapID, err)
 	}
 
 	// If the tree is empty and the map is empty,
 	// add the empty map root to the log.
 	if logRoot.GetSignedLogRoot().GetTreeSize() == 0 &&
 		mapRoot.GetMapRoot().GetMapRevision() == 0 {
 		glog.Infof("Initializing Trillian Log with empty map root")
-		if err := queueLogLeaf(ctx, s.tlog, s.logID, mapRoot.GetMapRoot()); err != nil {
+		if err := queueLogLeaf(ctx, s.tlog, logID, mapRoot.GetMapRoot()); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// StartSequencingAll startes sequencing processes for all domains.
+func (s *Sequencer) StartSequencingAll(ctx context.Context) error {
+	domains, err := s.admin.List(ctx, false)
+	if err != nil {
+		return fmt.Errorf("admin.List(): %v", err)
+	}
+	for _, d := range domains {
+		go s.StartSigning(ctx, d.LogID, d.MapID, d.MinInterval, d.MaxInterval)
+	}
+	// TODO(gbelvin): should we block until we return?
```

Comment:
What are the considerations?  Are you blocking in the function that calls this instead?
You might want to use a WaitGroup if you decide to block here.